packages/backend/src/open_payments/payment/outgoing/lifecycle.ts

```diff
@@ -146,7 +146,10 @@ export async function handleSending(
   const newMaxSourceAmount = payment.sendAmount.amount - amountSent
 
   let newMinDeliveryAmount
-  if (payment.receivingAccount) {
+  if (
+    payment.receivingAccount ||
+    !destination.destinationPaymentDetails?.incomingAmount
```

Comment:
```suggestion
    (destination.destinationPaymentDetails && !destination.destinationPaymentDetails.incomingAmount)
```
I think my original suggestion made the `MissingIncomingPayment` check below always pass.